src/server/endpoint.cc

```diff
@@ -208,7 +206,7 @@ namespace Pistache
         void Endpoint::init(const Endpoint::Options& options)
         {
             listener.init(options.threads_, options.flags_, options.threadsName_);
-            listener.setTransportFactory([&] {
+            listener.setTransportFactory([this, options] {
```

Comment:
@Tachi107 Yes, `options` must be captured by value because when transport factory will be called, `options` that was passed to  `Endpoint::init` can be deleted.